qiskit/qasm2/export.py

```diff
@@ -165,25 +167,53 @@ def dumps(circuit: QuantumCircuit, /) -> str:
         f"{'qreg' if isinstance(reg, QuantumRegister) else 'creg'} {name}[{reg.size}];"
         for name, reg in register_escaped_names.items()
     )
-    instruction_calls = []
-    for instruction in circuit._data:
-        operation = instruction.operation
-        if operation.name == "measure":
+
+    def qasm_line(instruction) -> str | None:
+        if instruction.is_control_flow():
+            operation = instruction.operation
+            if operation.name != "if_else":
+                raise QASM2ExportError("OpenQASM 2 does not support control-flow constructs")
```

Comment:
That's not strictly true is it? 
```suggestion
                raise QASM2ExportError(
                    "OpenQASM 2 does not support control-flow constructs outside if_else"
                )
```